TYPO3.Flow/Classes/TYPO3/Flow/Reflection/ObjectAccess.php

```diff
@@ -394,6 +419,16 @@ public static function isPropertyGettable($object, $propertyName)
     }
 
     /**
+     * Check if the given $value is any type that can be assigned to a collection type property.
+     *
+     * @return boolean TRUE if $value is either NULL, an array or an instance of \Traversable
+     */
+    public static function isCollectionTypePropertyValue($value)
```

Comment:
Well, isCollectionType() currently only checks some very specific class names (and their children), while this should IMO also work with any class of type \Traversable, so we'd probably need to change TypeHandling for that. Plus - and that is more problematic - we also need to handle the case that the propertyValue is null (i.e. the whole collection is unset). So we would need to use reflection to get the actual annotated type of the property (slow) or still revert to considering **null** a collection type value. I'd go for the latter.